lineage/figures/figure3.py

```diff
@@ -41,140 +37,85 @@ def makeFigure():
 def accuracy():
     """
     Calculates accuracy and parameter estimation
-    over an increasing number of lineages in a population for
-    a uncensored two-state model.
-    We increase the desired number of cells in a lineage by
-    the experiment time.
+    over an increasing number of cells in a lineage for
+    a uncensored two-state model but differing state distribution.
+    We vary the distribution by
+    increasing the Wasserstein divergence between the two states.
     """
 
     # Creating a list of populations to analyze over
-    num_lineages = np.linspace(min_num_lineages, max_num_lineages, num_data_points, dtype=int)
+    list_of_Es = [[StateDistribution(0.99, 0.8, 12, a, 10, 5), StateDistribution(0.99, 0.75, 12, 1, 9, 4)] for a in np.linspace(1, 10, 40)]
     list_of_populations = []
     list_of_fpi = []
     list_of_fT = []
     list_of_fE = []
-    for num in num_lineages:
+    for E in list_of_Es:
         population = []
 
-        for _ in range(num):
+        good2go = False
+        while not good2go:
+            tmp_lineage = LineageTree(pi, T, E, max_desired_num_cells)
+            good2go = lineage_good_to_analyze(tmp_lineage)
 
-            good2go = False
-            while not good2go:
-                tmp_lineage = LineageTree(pi, T, E2, min_desired_num_cells)
-                good2go = lineage_good_to_analyze(tmp_lineage)
-
-            population.append(tmp_lineage)
+        population.append(tmp_lineage)
 
         # Adding populations into a holder for analysing
         list_of_populations.append(population)
         list_of_fpi.append(pi)
         list_of_fT.append(T)
-        list_of_fE.append(E2)
-
-    return commonAnalyze(list_of_populations)
-
-
-def figureMaker2(ax, x, paramEst, accuracies, tr, pii, paramTrues, xlabel="Number of Cells"):
+        list_of_fE.append(E)
+
+    wass, _, Accuracy, _, _, paramTrues, _ = commonAnalyze(list_of_populations, xtype="wass")
+    total = []
+    for i in range(4):
+        tmp1 = list(sp.gamma.rvs(a=paramTrues[i, 0, 3], loc=0.0,
+                                 scale=paramTrues[i, 0, 5], size=200))
+        total.append(tmp1)
+        tmp2 = list(sp.gamma.rvs(a=paramTrues[i, 1, 3], loc=0.0,
+                                 scale=paramTrues[i, 1, 5], size=200))
+        total.append(tmp2)
```

Comment:
What is this for loop for?

Reply:
Nice catch! I found the reason why the violin plots were not showing different distributions!